lib/mongoose_push/service/apns/certificate.ex

```diff
@@ -0,0 +1,46 @@
+defmodule MongoosePush.Service.APNS.Certificate do
+  @moduledoc false
+
+  require Record
+  Record.defrecord :otp_cert,
+    Record.extract(:OTPCertificate, from_lib: "public_key/include/public_key.hrl")
+  Record.defrecord :tbs_cert,
+    Record.extract(:TBSCertificate, from_lib: "public_key/include/public_key.hrl")
+  Record.defrecord :cert_ext,
+    Record.extract(:Extension,      from_lib: "public_key/include/public_key.hrl")
+
+  # From: https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingwithAPNs.html
+  @apns_topic_extn_id {1, 2, 840, 113635, 100, 6, 3, 6}
+
+  def extract_topics!(cert_file) do
+    maybe_extension =
+      cert_file
+      |> File.read!()
+      |> :public_key.pem_decode()
+      |> List.keyfind(:Certificate, 0)
+      |> get_cert_extension!(@apns_topic_extn_id)
+
+    if maybe_extension == nil do
+        throw :no_extension
```

Comment:
An "elixir error" (raise "sth") won't be catched by your `try do catch` in `maybe_setup_default_apns_topic`. It will only catch the `throw`. If it's not different than any other error then it should case a whole thing to fail - as any other error would do.

I still see inconsistency here.